packages/context-editor/src/components/AttributePanel.tsx

```diff
@@ -1,110 +1,93 @@
 import type { Mark } from "prosemirror-model";
 
-import React, { useEffect, useState } from "react";
-import { EditorView } from "prosemirror-view";
+import React from "react";
+import { usePluginViewContext } from "@prosemirror-adapter/react";
+import { useDebouncedCallback } from "use-debounce";
 
 import { Input } from "ui/input";
 import { Label } from "ui/label";
 
-import type { PanelProps } from "../ContextEditor";
+import { attributePanelKey } from "../plugins/attributePanel";
+import { baseSchema } from "../schemas";
+import { InlineLinkMenu } from "./InlineLinkMenu";
 
 const animationTimeMS = 150;
 const animationHeightMS = 100;
 
-export interface AttributePanelProps {
-	panelPosition: PanelProps;
-	viewRef: React.RefObject<EditorView | null>;
-}
+export function AttributePanel() {
```

Comment:
i made smaller changes to this component than were probably necessary:
- got rid of its local state and used the parent's (maybe a mistake?). ideally it wouldn't depend on the `node` in the parent component state being up to date at all and could instead use prosemirror as the source of truth (perhaps just from the selection, or maybe directly in the plugin state). it should probably also submit all of its values at once instead of using these onchange handlers on individual inputs.
- moved it inline in the doc (also maybe a mistake)
- conditionally render the special link menu (not really functional, i wouldn't worry about it)